# Exploratory Data Analysis
- Zach Hanson

## Libraries

In [1]:
#Standard libraries
import pandas as pd
import numpy as np

## Loading in Data

In [2]:
#Loading and checking both years
df_2000 = pd.read_csv('Data/final_tmdb_data_2000.csv.gz', 
                       low_memory = False)
df_2000.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ...     revenue  runtime  \
0       NaN               NaN              NaN  ...         NaN      NaN   
1   62127.0                en  The Fantasticks  ...         0.0     86.0   
2  110977.0                en    For the Cause  ...         0.0    100.0   
3  442869.0                hi             Gang  ...         0.0    152.0   
4     843.0                cn             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.112     2135.0            PG  

[5 rows x 26 columns]

In [3]:
df_2000.shape

(1222, 26)

- 1222 Movies from 2000

In [4]:
df_2001 = pd.read_csv('Data/final_tmdb_data_2001.csv.gz',
                       low_memory = False)
df_2001.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /hfeiSfWYujh6MKhtGTXyK3DD4nN.jpg                   NaN   
2  tt0114447    0.0                               NaN                   NaN   
3  tt0118589    0.0  /9NZAirJahVilTiDNCHLFcdkwkiy.jpg                   NaN   
4  tt0118652    0.0  /mWxJEFRMvkG4UItYJkRDMgWQ08Y.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 10749, 'name': 'Romance'}, {'id': 14, ...      NaN   
2         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3  22000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...      NaN   
4   1000000.0  [{'id': 27, 'name': 'Horror'}, {'id': 9648, 'n...      NaN   

         id original_language         original_title  ...     revenue  \
0       NaN               NaN                    NaN  ...         NaN   
1   11232.0                en         Kate & Leopold  ...  76019048.0   
2  151007.0                en       The Silent Force  ...         0.0   
3   10696.0                en                Glitter  ...   5271666.0   
4   17140.0                en  The Attic Expeditions  ...         0.0   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1    118.0  [{'english_name': 'French', 'iso_639_1': 'fr',...  Released   
2     90.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3    104.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4    100.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                  title  \
0                                                NaN                    NaN   
1  If they lived in the same century, they'd be p...         Kate & Leopold   
2  They left him for dead... They should have fin...       The Silent Force   
3   In music she found her dream, her love, herself.                Glitter   
4  His search for peace of mind... will leave his...  The Attic Expeditions   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         6.325     1177.0         PG-13  
2    0.0         5.000        3.0           NaN  
3    0.0         4.600      122.0         PG-13  
4    0.0         5.100       29.0             R  

[5 rows x 26 columns]

In [5]:
df_2001.shape

(1305, 26)

- 1305 Movies from 2001

In [6]:
#Concatenate to one dataframe
dfs = [df_2000, df_2001]
merged_df = pd.concat(dfs)
merged_df.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ...     revenue  runtime  \
0       NaN               NaN              NaN  ...         NaN      NaN   
1   62127.0                en  The Fantasticks  ...         0.0     86.0   
2  110977.0                en    For the Cause  ...         0.0    100.0   
3  442869.0                hi             Gang  ...         0.0    152.0   
4     843.0                cn             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.112     2135.0            PG  

[5 rows x 26 columns]

In [7]:
merged_df.shape

(2527, 26)

- 2527 Movies total after merging 2000 and 2001
    - This is the number we expect so they merged properly

In [8]:
#Saving the merged dataframe to a csv file
merged_df.to_csv('Data/tmdb_results_combined.csv.gz', compression='gzip', index=False)

## Movies with Valid Financial Information

In [9]:
#Filling missing values with "0" so we can explore
merged_df['budget'] = merged_df['budget'].fillna(0)
merged_df['revenue'] = merged_df['revenue'].fillna(0)

merged_df.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres homepage  \
0         0.0                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ...     revenue  runtime  \
0       NaN               NaN              NaN  ...         0.0      NaN   
1   62127.0                en  The Fantasticks  ...         0.0     86.0   
2  110977.0                en    For the Cause  ...         0.0    100.0   
3  442869.0                hi             Gang  ...         0.0    152.0   
4     843.0                cn             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.112     2135.0            PG  

[5 rows x 26 columns]

In [10]:
#Finding how many movies have some valid financial information
valid_fin_info = merged_df.loc[((merged_df['budget'] > 0) |
                                (merged_df['revenue'] > 0))]
valid_fin_info.shape

(632, 26)

- There is 632 movies with at least some valid financial info for years 2000 and 2001

In [11]:
#Excluding Movies with 0 for budget AND revenue for remaining analysis
filter_df = merged_df.loc[~((merged_df['budget'] == 0.0) & 
                            (merged_df['revenue'] == 0.0))]

## Movies per Cert Category

In [12]:
filter_df['certification'].value_counts()

R        230
PG-13    131
PG        36
NR        17
G         14
Name: certification, dtype: int64

## Average Revenue per Cert Category

In [13]:
filter_df.groupby('certification')['revenue'].mean()

certification
G        1.237463e+08
NR       9.588674e+06
PG       1.095338e+08
PG-13    9.926299e+07
R        3.306692e+07
Name: revenue, dtype: float64

## Average Budget per Cert Category

In [14]:
filter_df.groupby('certification')['budget'].mean()

certification
G        4.085714e+07
NR       6.302358e+06
PG       4.381937e+07
PG-13    4.300884e+07
R        1.966089e+07
Name: budget, dtype: float64